app/models/setting.rb

```diff
@@ -13,4 +13,17 @@ def self.get(key)
     value_object = Setting.where(key: key, status: "Active").first
     value_object.value if value_object
   end
+
+  def active?
+    self.status == 'Active'
+  end
+  
+  # this method should be used only for the settings that have integer values
+  # converts the given string to integer, if not possible returns nil.
+  def self.get_integer(key, fallback_value = nil)    
+    value = get(key)
+    (Integer(value) rescue nil) || fallback_value
```

Comment:
Still struggling with this syntax some, and so is Code Climate. I tried it out in a console and it threw an error. What would you think of `value.is_a?(Integer)`

Reply:
yeah, that should also work, 
`Integer(value) rescue nil` is somewhat short. 
Code clarity is more important, I will change it.

Reply:
sorry @kgf, I think it wont work, because the type of the column, "value" is text, so 
`setting.value` will return the value in string format (example, '9'), so `value.is_a?(Integer)` will return false.
I cant think of any other way other than `Integer(value) rescue nil` as if the value can be converted it will convert and int value will be returned, if it cant, raised exception will be rescued and nil be returned.

We can use `value.to_i.to_s == value` but it wont work in some case like when value is '01'.
Or we can use regex not sure about that too.

Please let me know, if you have any better solution for this.

Reply:
Phone a friend... BRB. Thanks for your diligence on this !

Reply:
presuming `get(key)` always returns a string... if it returns nil you'll have to add a check for that too .

```
def self.get_integer(key, fallback_value = nil)
  raw_value = get(key)
  value = raw_value.to_i 
  if value == 0 and !raw_value.match(/^\s*0+(\.0+)?\s*$/)
      value = fallback_value
  end
  value
end
```

Reply:
Wouldn’t it be better if we just use `Integer(value) rescue fallback_value`.
@kgf @masukomi what do you think ?, let me know your thoughts.

Reply:
Yeah, @yasirazgar I think you're right. Your line is atypical but i think it's relatively reasonable.

Reply:
Yes, you've convinced me too. I don't care for @masukomi solution as much, because it looks like you could not have a setting of zero, as it would think that meant no setting.
Thanks to both of you for your work on this !